### PR TITLE
Made UserDataTransformer configurable allowing to inject different im…

### PR DESCRIPTION
**Pull request by benatespina**
…plementations

### CODE REVIEW

File: src/BenGorUser/UserBundle/DependencyInjection/Compiler/Application/DataTransformer/UserDTODataTransformerBuilder.php [hunk 1]
```diff
@@ -39,14 +40,23 @@ class UserDTODataTransformerBuilder implements ApplicationBuilder
...
+     */
+    protected $dataTransformer;
+
+    /**
      * Constructor.
      *
-     * @param ContainerBuilder $container     The container builder
-     * @param array            $configuration The configuration tree
+     * @param ContainerBuilder $container       The container builder
+     * @param string           $dataTransformer The FQCN of user data transformer
```
Comment (gorkalaucirica) [inline src/BenGorUser/UserBundle/DependencyInjection/Compiler/Application/DataTransformer/UserDTODataTransformerBuilder.php]: "or service name"

Comment (benatespina) [inline src/BenGorUser/UserBundle/DependencyInjection/Compiler/Application/DataTransformer/UserDTODataTransformerBuilder.php]: service id

Comment (gorkalaucirica) [inline src/BenGorUser/UserBundle/DependencyInjection/Compiler/Application/DataTransformer/UserDTODataTransformerBuilder.php]: 👌

**Comment by gorkalaucirica** [comment]
We should add to docs how to extend UserDTOTransformer and how to change it in the config.yml

**Comment by gorkalaucirica** [comment]
@benatespina docs added

**Pull request merged**